perf/bps_tree.cc

```diff
@@ -0,0 +1,464 @@
+#include <stdint.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <inttypes.h>
+#include <time.h>
+#include <random>
+
+#include <benchmark/benchmark.h>
+#include <trivia/util.h>
+
+/* A simple test tree. */
+
+#define tree_i64_EXTENT_SIZE 2048
+#define tree_i64_elem_t int64_t
+#define tree_i64_key_t int64_t
+
+#define BPS_TREE_NAME tree_i64_t
+#define BPS_TREE_BLOCK_SIZE 512
```

Comment:
Regarding your benchmark extension recipe, I don't think it's feasible to create other tree configurations right now, as it would require copy-pasting the benchmarks for each tree configuration, AFAIU. 

We should either generate all the benchmarks for a given a tree configuration parameter with a macro, or we should just leave the tree configuration as high-level benchmark parameters, which can be tweaked manually.

For the second way, I would organize the parameters somehow like this (with macros instead of `constexpr`s, where needed):
https://github.com/tarantool/tarantool/blob/94d2813841a649d5254974dd921aed01af1b24d4/perf/memtx.cc#L43-L70